cmd/snap/cmd_set.go

```diff
@@ -155,6 +161,48 @@ func (x *cmdSet) Execute([]string) error {
 	return nil
 }
 
+func validateJsonKeys(v interface{}) error {
+	var initial map[string]interface{}
+	var ok bool
+
+	if initial, ok = v.(map[string]interface{}); !ok {
+		// If it's not a map, there's nothing to validate.
+		// Because, there is no nested map or array.
+		return nil
+	}
+
+	stack := []map[string]interface{}{initial}
+
+	for len(stack) > 0 {
+		current := stack[len(stack)-1]
+		stack = stack[:len(stack)-1]
+
+		for key, value := range current {
+			if err := config.ValidateKey(key); err != nil {
+				return err
+			}
+
+			handleNestedMap(value, &stack)
+			handleNestedArray(value, &stack)
+		}
+	}
+	return nil
+}
+
+func handleNestedMap(v interface{}, stack *[]map[string]interface{}) {
+	if nestedMap, ok := v.(map[string]interface{}); ok {
+		*stack = append(*stack, nestedMap)
+	}
+}
+
+func handleNestedArray(v interface{}, stack *[]map[string]interface{}) {
+	if nestedArray, ok := v.([]interface{}); ok {
+		for _, nestedValue := range nestedArray {
+			handleNestedMap(nestedValue, stack)
+		}
+	}
+}
```

Comment:
As these are only called once, I don't think they need to be split out into their own functions, particularly with passing a pointer to a slice. Just put a one-line comment before each if statement in-place.

Reply:
Sure! Good point! As I'm refactoring this, these two functions will go away.